### ISSUE TITLE
Node Pool keep on getting created and there is no exit in terraform.

### ISSUE THREAD

**Issue by anupam-sy** [terraform-google-modules/terraform-google-kubernetes-engine#1608]
### TL;DR

Ideally the below mentioned terraform configuration should create the Private GKE cluster and the node pool. This code creates the cluster successfully but it is unable to create the node pools. Actually nodes are getting created when checked in compute engine section, but those node pool shows in provisioning mode only and terraform output also shows "creating node pool". Terraform output is below and also Added the log output in observed behavior section.

module.gke_cluster_02.google_container_node_pool.pools["node-pool-01"]: Still creating... [10m40s elapsed]
module.gke_cluster_02.google_container_node_pool.pools["node-pool-01"]: Still creating... [10m50s elapsed]
module.gke_cluster_02.google_container_node_pool.pools["node-pool-01"]: Still creating... [11m0s elapsed]
module.gke_cluster_02.google_container_node_pool.pools["node-pool-01"]: Still creating... [11m10s elapsed]
module.gke_cluster_02.google_container_node_pool.pools["node-pool-01"]: Still creating... [11m20s elapsed]
module.gke_cluster_02.google_container_node_pool.pools["node-pool-01"]: Still creating... [11m30s elapsed]
module.gke_cluster_02.google_container_node_pool.pools["node-pool-01"]: Still creating... [11m40s elapsed]
module.gke_cluster_02.google_container_node_pool.pools["node-pool-01"]: Still creating... [11m50s elapsed]
module.gke_cluster_02.google_container_node_pool.pools["node-pool-01"]: Still creating... [12m0s elapsed]
module.gke_cluster_02.google_container_node_pool.pools["node-pool-01"]: Still creating... [12m10s elapsed]
module.gke_cluster_02.google_container_node_pool.pools["node-pool-01"]: Still creating... [12m20s elapsed]
module.gke_cluster_02.google_container_node_pool.pools["node-pool-01"]: Still creating... [12m30s elapsed]
module.gke_cluster_02.google_container_node_pool.pools["node-pool-01"]: Still creating... [12m40s elapsed]
module.gke_cluster_02.google_container_node_pool.pools["node-pool-01"]: Still creating... [12m50s elapsed]
module.gke_cluster_02.google_container_node_pool.pools["node-pool-01"]: Still creating... [13m0s elapsed]
module.gke_cluster_02.google_container_node_pool.pools["node-pool-01"]: Still creating... [13m10s elapsed]
module.gke_cluster_02.google_container_node_pool.pools["node-pool-01"]: Still creating... [13m20s elapsed]

### Expected behavior

Ideally this should create the Private GKE cluster and the provided node pool/pools.

### Observed behavior

-----------------------------------------------------: timestamp=2023-04-17T12:23:37.955+0530
2023-04-17T12:23:38.055+0530 [INFO]  provider.terraform-provider-google_v4.51.0_x5: 2023/04/17 12:23:38 [DEBUG] Google API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 200 OK
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Date: Mon, 17 Apr 2023 06:53:38 GMT
Server: ESF
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

{
 "name": "operation-1681713882753-92c131e1-9a1b-4da3-94dd-b12acb8b0bca",
 "zone": "asia-southeast1",
 "operationType": "CREATE_NODE_POOL",
 "status": "RUNNING",
 "selfLink": "https://container.googleapis.com/v1/projects/283971982734/locations/asia-southeast1/operations/operation-1681713882753-92c131e1-9a1b-4da3-94dd-b12acb8b0bca",
 "targetLink": "https://container.googleapis.com/v1/projects/283971982734/locations/asia-southeast1/clusters/pri-cluster-02/nodePools/node-pool-01",
 "startTime": "2023-04-17T06:44:42.75302003Z"
}

### Terraform Configuration

```hcl
`module "gke_cluster_02" {
  source             = "terraform-google-modules/kubernetes-engine/google//modules/private-cluster"
  project_id         = var.project_id
  name               = "pri-cluster-02"
  regional           = true
  region             = var.shared_resource_region
  zones              = ["asia-southeast1-a", "asia-southeast1-b", "asia-southeast1-c"]
  release_channel    = "STABLE"
  kubernetes_version = "1.24.10-gke.2300"

  network_project_id = var.shared_network_project_id
  network            = data.google_compute_network.shared_network.name
  subnetwork         = data.google_compute_subnetwork.shared_subnetwork.name
  ip_range_pods      = data.google_compute_subnetwork.shared_subnetwork.secondary_ip_range[0].range_name
  ip_range_services  = data.google_compute_subnetwork.shared_subnetwork.secondary_ip_range[1].range_name

  enable_private_endpoint = true
  enable_private_nodes    = true
  master_ipv4_cidr_block  = "176.16.0.0/28"

  create_service_account   = true
  remove_default_node_pool = true

  add_cluster_firewall_rules = true
  firewall_inbound_ports     = ["9443", "15017"]

  master_authorized_networks = [
    {
      cidr_block   = "10.0.91.0/24"
      display_name = "vpc"
    },
  ]

  node_pools = [
    {
      name            = "node-pool-01"
      machine_type    = "e2-small"
      min_count       = 1
      max_count       = 2
      disk_size_gb    = 50
      disk_type       = "pd-standard"
      image_type      = "COS_CONTAINERD"
      auto_repair     = true
      auto_upgrade    = true
      preemptible     = false
    },
  ]
}`
```

### Terraform Version

```sh
terraform {
  required_version = "~> 1.0.0"

  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "~> 4.51.0"
    }
    google-beta = {
      source  = "hashicorp/google-beta"
      version = "~> 4.51.0"
    }
  }
}
```
